### ISSUE TITLE
[Overview] Good things removed from Nixpkgs/NixOS

### ISSUE THREAD

**Issue by oxij** [SLNOS/nixpkgs#1]
An overview issue for tracking the contents of `it-aint-broken` SLNOS public patchset (not broken useful things that were removed from NixOS/nixpkgs without adequate replacements).

## Currently contains

- SLiM and nixos/auto display managers (NixOS/nixpkgs#78583, NixOS/nixpkgs#73251, a part of NixOS/nixpkgs#30890)
- old Firefox variants, IceCat, Tor Browser (NixOS/nixpkgs#79115, NixOS/nixpkgs#77507, NixOS/nixpkgs#77452)

## Current candidates

- Older kernels
- Older gcc versions

## How to propose new candidates

Just point me to a relevant commit. If it seems useful and easy to maintain, I'll add it. If it seems hard to maintain, we will discuss it.

**Comment by volth** [comment]
Ironically, I have been affected exactly by the same 2 changes :)

BTW, firefox-60 needs rust-1.37, which has been removed from `master` and `release-20.03`, so if you are going to rebase your branch, it will diverge even further.

**Comment by oxij** [comment]
Yep, because those are, IMHO, two the most insane removals to date.

> rebase will diverge even further

Yep, not a problem, SLNOS patchsets contain hundreds of commits, so we have tooling for automating this stuff (as, I suspect, upstream tor-browser developers also do for firefox, because rebasing more than 10 commits at once while fixing conflicts with pure git without going insane is just too hard).

I'm just too lazy to sit down and filter out all the sensible things and publish them (with some sensible DIY-capable tooling for this, or, possibly, I just need to automate the publishing of automated-rebase-plus-eval-checked results, though I'm a bit paranoid about the later, for technical reasons).

**Comment by volth** [comment]
I restored old `firefox`es to `firefox-esr` directory to minimize interference with ongoing development in `firefox-70+`

`firefox-esr-52` and `firefox-esr-60` also need new patches for recently introduced `glibc-2.30`, so it would be simpler to maintain old `firefox`es as a separate project than as spaghetti of `if versionAtLeast`

https://github.com/volth/nixpkgs-windows/commit/ed430d3e4e8a873938f99212230264e6b718055b It is based on `release-20.03`

**Comment by volth** [comment]
Unfortunately, it is difficult to maintain in NUR because of changes in Rust
